### PR TITLE
Remove unused versioned_client

### PR DESCRIPTION
**Pull request by wackxu**
**What this PR does / why we need it**:

These function is never used and We do not suggest use the internal client version in `kubectl`. So we can remove it safely.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/kubectl/versioned_client.go [hunk 1]
```diff
@@ -1,39 +0,0 @@
...
-package kubectl
-
-import (
-	clientappsv1beta1 "k8s.io/client-go/kubernetes/typed/apps/v1beta1"
-	clientextensionsv1beta1 "k8s.io/client-go/kubernetes/typed/extensions/v1beta1"
-	internalclientset "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
-)
-
-// TODO: get rid of this and plumb the caller correctly
-func versionedExtensionsClientV1beta1(internalClient internalclientset.Interface) clientextensionsv1beta1.ExtensionsV1beta1Interface {
```
Comment (shiywang) [inline pkg/kubectl/versioned_client.go]: could you point me the commit of when this file first time introduced so I can check if it's necessary to remove it ?

Comment (wackxu) [reply pkg/kubectl/versioned_client.go]: @shiywang first commit https://github.com/kubernetes/kubernetes/commit/96cd71d8f64bb7897f18115f46986f514a0722f3#diff-0bb9c459464f4ed2c48919a97c4cd8e6

Comment (wackxu) [reply pkg/kubectl/versioned_client.go]: kindly ping @shiywang for review

**Pull request closed without merging**